packages/instantsearch.js/src/components/Template/Template.tsx

```diff
@@ -10,6 +10,28 @@ import type { BindEventForHits, SendEventForHits } from '../../lib/utils';
 import type { Templates } from '../../types';
 import type { JSX } from 'preact';
 
+class RawHtml extends Component<{ content: string }> {
+  ref = createRef();
+  nodes: Element[] = [];
+
+  componentDidMount() {
+    const fragment = new DocumentFragment();
+    const root = document.createElement('div');
+    root.innerHTML = this.props.content;
+    this.nodes = [...root.children];
+    fragment.append(...this.nodes);
+    this.ref.current.replaceWith(fragment);
+  }
+
```

Comment:
do we not need a componentWill/HaveReceiveProps in case content changes?

Reply:
There is but we still need to unmount anyway, not sure if it's worth spending so much time on something caused by tech debt.
Only win would be slightly better performance for those who still use string templates which we discourage 🤷‍♂️

Reply:
makes sense, was just not sure if it's a bug or if it properly updates :)